### ISSUE TITLE
Tree entities don't work properly with RelationIdLoader

### ISSUE THREAD

**Issue by josephktcheung** [typeorm/typeorm#2042]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

Tree entities cannot be fetched properly by using `RelationIdLoader`. Here's the reproduction branch. https://github.com/josephktcheung/test-typeorm/tree/feature/tree_entities_relationIdLoader

Here's my tree entity setting: category can have multiple child categories
```ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, OneToOne, JoinColumn, ManyToOne, ManyToMany, JoinTable, Tree, TreeParent, TreeChildren } from 'typeorm';

@Entity()
export class Category {

  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @ManyToOne(() => Category, category => category.children)
  parent: Category;

  @OneToMany(() => Category, category => category.parent)
  children: Category[];
}
```

Here's how I set up the entity relationship.
```ts
  const categoryRepository = connection.getRepository(Category);

  const categoryParent = new Category();
  categoryParent.name = 'category parent';

  await categoryRepository.save(categoryParent);
  const categoryChild = new Category();

  categoryChild.name = 'category children';
  categoryChild.parent = categoryParent;

  await categoryRepository.save(categoryChild);
```

If I call `RelationIdLoader`'s `loadManyToManyRelationIdsAndGroup` with categoryParent like this:
```ts

  connection.entityMetadatas.forEach(entityMetadata => {
    entityMetadata.relations.forEach(relation => {
      promises.push(connection.relationIdLoader.loadManyToManyRelationIdsAndGroup(relation, [categoryParent]));
    })
  });

```

the result is:
```ts
[
  [
    {
      "entity": {
        "name": "category parent",
        "id": 1
      }
    }
  ],
  [
    {
      "entity": {
        "name": "category parent",
        "id": 1
      },
      "related": []
    }
  ]
]
```
Where you can see that the related array is empty. We expect to have something like this instead:

```ts
[
  [
    {
      "entity": {
        "name": "category parent",
        "id": 1
      },
      "related": {
        "id": 1,
        "name": "category parent"
      }
    }
  ],
  [
    {
      "entity": {
        "name": "category parent",
        "id": 1
      },
      "related": [
           {
               "id": 2,
               "name": "category child"
           }
       ]
    }
  ]
]
```

And here's a problematic sql statement generated by typeorm:
```sql
query: SELECT "Category"."id" AS "Category_id", "Category"."parentId" AS "Category_id" FROM "category" "Category" WHERE "Category"."parentId" IN ($1) -- PARAMETERS: [1]
```
You can see that when selecting primary column and join column ids, the join column id is selected as `"Category_id"` which is the same as the primary column alias.

This bug stops Vesper from fetching tree entities correctly, for details please read https://github.com/vesper-framework/vesper/issues/16

**Comment by pleerock** [comment]
I faced issue with self-referencing entities recently as well, and fixed this bug. I'll publish 0.2.4 please check if it resolves this issue as well.

**Comment by josephktcheung** [comment]
BTW @pleerock it'd be nice to have typeorm used in vesper upgraded after 0.2.4 is released, thanks in advance :)

**Comment by pleerock** [comment]
vesper should use `^` version of typeorm, so you should have latest once you install vesper from scratch again. I released `0.2.4` please check it.
